crunchy.go

```diff
@@ -142,12 +143,27 @@ func (v *Validator) foundInDictionaries(s string) error {
 	// find mangled / reversed passwords
 	// we can skip this if the pw is longer than the longest word plus our minimum distance
 	if pwlen <= v.wordsMaxLen+v.options.MinDist {
+		jobs := runtime.GOMAXPROCS(0)
+		queue := make(chan struct{}, jobs)
+		errc := make(chan *DictionaryError, jobs)
 		for word := range v.words {
-			if dist := smetrics.WagnerFischer(word, pw, 1, 1, 1); dist <= v.options.MinDist {
-				return &DictionaryError{ErrMangledDictionary, word, dist}
-			}
-			if dist := smetrics.WagnerFischer(word, revpw, 1, 1, 1); dist <= v.options.MinDist {
-				return &DictionaryError{ErrMangledDictionary, word, dist}
+			select {
+			case queue <- struct{}{}:
```

Comment:
I'm not sure I fully understand the purpose of this queue. Couldn't we just keep launching a couple of worker threads in a pool and feed them with all the words from the wordlist? I think it may turn out a bit more readable. Let me know if you already tried that but this queue simply outperforms it :smiley:

Reply:
I tried some approaches, e.g. using `x/sync/errgroup`, and this was the simplest way I found to make it concurrency bounded with the ability to cancel on the first error found.

The 'queue' is just a semaphore that keeps us from spawning more than 'njobs' goroutines simultaneously.